core/store/src/migrations.rs

```diff
@@ -358,3 +360,39 @@ pub fn migrate_39_to_40(store: &Store) -> anyhow::Result<()> {
     update.commit()?;
     Ok(())
 }
+
+/// Migrates the database from version 40 to 41.
+///
+/// This rewrites the contents of the StateDlInfos column
+pub fn migrate_40_to_41(store: &Store) -> anyhow::Result<()> {
+    #[derive(BorshSerialize, BorshDeserialize)]
+    struct LegacyStateSyncInfo {
+        sync_hash: CryptoHash,
+        shards: Vec<ShardInfo>,
+    }
+
+    let mut update = store.store_update();
+
+    for row in store.iter_prefix_ser::<LegacyStateSyncInfo>(DBCol::StateDlInfos, &[]) {
```

Comment:
mini nit: it's a bit funny to call `iter_prefix` and use empty prefix - if it's simple enough I would suggest implementing `iter_ser` (assuming it's missing).

Reply:
implemented and also updated one other caller in this migrations.rs file